### PR TITLE
fix vanilla premium items going over 140k

### PR DESCRIPTION
**Pull request by qndel**
resolves https://github.com/diasurgical/devilutionX/issues/1478
turns out wirt had the same bug so fixed it too + changed the stat ifs into max
+wirt price bug
resolves https://github.com/diasurgical/devilutionX/issues/1480

### CODE REVIEW

File: Source/items.cpp [hunk 1]
```diff
@@ -4664,8 +4655,10 @@ static void SpawnOnePremium(int i, int plvl, int myplr)
 		GetItemBonus(0, itype, plvl >> 1, plvl, true, !gbIsHellfire);
 
 		if (!gbIsHellfire) {
-			if (items[0]._iIvalue > 140000)
+			if (items[0]._iIvalue > 140000) {
+				keepgoing = true; // prevent breaking the do/while loop too early by failing hellfire's condition in while
```
Comment (qndel) [inline Source/items.cpp:4659]: while this is a relatively ugly hack, I really think we should stop refactoring till we find and fix 1.2.0 issues :D

Comment (AJenbo) [reply Source/items.cpp:4659]: Are you saying that to get us to focus on looking in to the bug? We are not going to include the refactoring in 1.2.1 if that's your concern.

File: Source/items.cpp [hunk 2]
```diff
@@ -4994,8 +4977,10 @@ void SpawnBoy(int lvl)
 			GetItemBonus(0, itype, lvl, 2 * lvl, true, true);
 
 			if (!gbIsHellfire) {
-				if (items[0]._iIvalue > 140000)
+				if (items[0]._iIvalue > 140000) {
```
Comment (AJenbo) [inline Source/items.cpp:4980]: Shoudn't this be corrected also?
```suggestion
				if (items[0]._iIvalue > 90000) {
```

File: Source/items.cpp (hunk 1 above)
Comment (qndel) [reply Source/items.cpp:4659]: oh, didn't know :D

File: Source/items.cpp (hunk 2 above)
Comment (qndel) [reply Source/items.cpp:4980]: yes but I left it for another PR as it requires going through this function and figuring out if it matches the original :D and I wasn't feeling like it atm ;p

**Review by AJenbo — APPROVED** [review]

**Pull request merged**